### PR TITLE
Adding a validation webhook to the `ManagedClusterModule` CRD. (#368)

### PR DESCRIPTION
**Pull request by ybettan**
Following the webhook added for the `Module` CRD we now also need a validation webhook to validate the `ManagedClusterModule` which is used in the hub-spoke topology.

---

Upstream-Commit: 37475d3b257a9fa6275dc50b283d2990de87c316

### CODE REVIEW

File: config/webhook-hub/service.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+
+apiVersion: v1
+kind: Service
+metadata:
```
Comment (qbarrand) [inline config/webhook-hub/service.yaml:4]:
```suggestion
metadata:
  annotations:
    service.beta.openshift.io/serving-cert-secret-name: webhook-server-cert
```

File: config/webhook-hub/kustomizeconfig.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    path: webhooks/clientConfig/service/name
+
+namespace:
+- kind: ValidatingWebhookConfiguration
+  group: admissionregistration.k8s.io
+  path: webhooks/clientConfig/service/namespace
+  create: true
+
+varReference:
+- path: metadata/annotations
```
Comment (qbarrand) [inline config/webhook-hub/kustomizeconfig.yaml:18]:
```suggestion
- path: metadata/annotations

namePrefix:
- kind: Service
  name: webhook-service
  path: metadata/annotations/service.beta.openshift.io\/serving-cert-secret-name
```

File: config/default-hub/webhookcainjection_patch.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  labels:
+    app.kubernetes.io/name: validatingwebhookconfiguration
+    app.kubernetes.io/instance: validating-webhook-configuration
+    app.kubernetes.io/component: webhook
+    app.kubernetes.io/created-by: kernel-module-management
+    app.kubernetes.io/part-of: kernel-module-management
+    app.kubernetes.io/managed-by: kustomize
+  name: validating-webhook-configuration
+  annotations:
+    cert-manager.io/inject-ca-from: $(CERTIFICATE_NAMESPACE)/$(CERTIFICATE_NAME)
```
Comment (qbarrand) [inline config/default-hub/webhookcainjection_patch.yaml:16]:
```suggestion
    service.beta.openshift.io/inject-cabundle: 'true'
```

File: config/webhook-hub/service.yaml (hunk 1 above)
Comment (ybettan) [reply config/webhook-hub/service.yaml:3]: Fixed.

File: config/webhook-hub/kustomizeconfig.yaml (hunk 2 above)
Comment (ybettan) [reply config/webhook-hub/kustomizeconfig.yaml:18]: Fixed.

File: config/default-hub/webhookcainjection_patch.yaml (hunk 3 above)
Comment (ybettan) [reply config/default-hub/webhookcainjection_patch.yaml]: Fixed.

**Comment by ybettan** [comment]
/unhold

Tried to apply a `ManagedClusterModule` with no regexp or literal and I got:
```
root kernel-module-management-midstream (webhook) $ oc apply -f managedclustermodule.yaml
Error from server (failed to validate kernel mappings: regexp or literal must be set at kernelMappings[0]): error when creating "managedclustermodule.yaml": admission webhook "vmanagedclustermodule.kb.io" denied the request: failed to validate kernel mappings: regexp or literal must be set at kernelMappings[0]
```

**Comment by qbarrand** [comment]
@ybettan please append `Upstream-Commit: 37475d3b257a9fa6275dc50b283d2990de87c316` to the commit message.

**Comment by ybettan** [comment]
Done.
@qbarrand

**Pull request merged**